corehq/messaging/smsbackends/twilio/views.py

```diff
@@ -42,3 +44,46 @@ def ivr_in(request):
         return HttpResponse(IVR_RESPONSE)
     else:
         return HttpResponseBadRequest("POST Expected")
+
+
+class TwilioIncomingSMSView(NewIncomingBackendView):
+    urlname = 'twilio_sms'
+
+    @property
+    def backend_class(self):
+        return SQLTwilioBackend
+
+    def get(self, request, api_key, *args, **kwargs):
+        return HttpResponseBadRequest("POST Expected")
+
+    def post(self, request, api_key, *args, **kwargs):
+        message_sid = request.POST.get('MessageSid')
+        account_sid = request.POST.get('AccountSid')
+        from_ = request.POST.get('From')
+        to = request.POST.get('To')
+        body = request.POST.get('Body')
+        incoming_sms(
+            from_,
+            body,
+            SQLTwilioBackend.get_api_id(),
+            backend_message_id=message_sid,
+            backend_id=self.backend_couch_id
+        )
+        return HttpResponse(EMPTY_RESPONSE)
+
+
+class TwilioIncomingIVRView(NewIncomingBackendView):
+    urlname = 'twilio_ivr'
+
+    @property
+    def backend_class(self):
+        return SQLTwilioBackend
+
+    def get(self, request, api_key, *args, **kwargs):
+        return HttpResponseBadRequest("POST Expected")
```

Comment:
Django views will return `HttpResponseNotAllowed` on request methods for which there aren't corresponding view methods.  You can also specify a class attr `http_method_names = ['post']` with just the methods you'd like to support.
http://ccbv.co.uk/projects/Django/1.9/django.views.generic.base/View/